File: passenger-service/src/main/java/com/software/modsen/passengerservice/service/PassengerService.java
package com.software.modsen.passengerservice.service;

import com.software.modsen.passengerservice.entity.Passenger;

import java.util.List;

public interface PassengerService {
    Passenger getById(Long id);

    List<Passenger> getAll(Integer pageNumber, Integer pageSize, String sortBy, Boolean includeRestricted);

    Passenger save(Passenger passenger);

    Passenger update(Passenger passenger);

    Passenger changeRestrictionsStatus(Long id);
}

File: passenger-service/src/main/kotlin/com/software/modsen/passengerservice/service/PassengerService.kt
package com.software.modsen.passengerservice.service

import com.software.modsen.passengerservice.dto.request.PassengerChangeStatusRequest
import com.software.modsen.passengerservice.entity.Passenger

interface PassengerService {
    fun getById(id: Long): Passenger
    fun getAll(pageNumber: Int, pageSize: Int, sortBy: String, includeRestricted: Boolean): List<Passenger?>
    fun save(passenger: Passenger): Passenger
    fun update(passenger: Passenger): Passenger
    fun changeRestrictionsStatus(request: PassengerChangeStatusRequest): Passenger
}